apollo-federation/src/composition/mod.rs

```diff
@@ -57,8 +63,21 @@ pub fn validate_subgraphs(
     }
 }
 
+/// Perform validations that require information about all available subgraphs.
+pub fn pre_merge_validations(
+    _subgraphs: &[Subgraph<Validated>],
+) -> Result<(), Vec<FederationError>> {
+    panic!("pre_merge_validations is not implemented yet")
+}
+
 pub fn merge_subgraphs(
     _subgraphs: Vec<Subgraph<Validated>>,
 ) -> Result<Supergraph<Merged>, Vec<FederationError>> {
     panic!("merge_subgraphs is not implemented yet")
 }
+
+pub fn post_merge_validations(
```

Comment:
Is this necessary? I think Sachin already did this with his PR this week: https://github.com/apollographql/router/pull/7451

I think we just need to call `validate_merged_schema` instead of creating this function.

Reply:
Those are just stubs and yes they might be unnecessary. Once we start filling in those stubs we'll determine whether we need it or not.

Sachin's PR does not use the new typestate approach and we might want to refactor it to use it.